src/server/multi_svr.c

```diff
@@ -306,3 +308,31 @@ gen_svr_inst_id(void)
 
 	return svr_inst_id;
 }
+
+/**
+ * @brief	Calculate the index of the current server
+ *
+ * @param	void
+ *
+ * @return	int
+ * @retval	index of the server
+ * @retval	-1 if couldn't be determined
+ */
+int
+get_server_index(void)
+{
+	static int svridx = -1;
+
+	if (svridx == -1) {
```

Comment:
Instead of calculating here, set the Index when we find the self in init_msi.

Reply:
"set the Index when we find the self in init_msi."

I'm not sure I understand. Is there any advantage of calculating it in init_msi? I'll have to create a global if I calculate it in init_msi() so that get_server_index() can return it, I'd rather not go in that direction. How does it help?

Reply:
init_msi will change and follows a similar logic with https://github.com/openpbs/openpbs/pull/2253
so it will be easier to merge.
We can just call one function whenever PSI changes.
The variable is already static, the suggested change requires it to be in the file's scope.